SYCL/SubGroup/helper.hpp

```diff
@@ -169,5 +169,14 @@ bool core_sg_supported(const device &Device) {
   auto Vec = Device.get_info<info::device::extensions>();
   if (std::find(Vec.begin(), Vec.end(), "cl_khr_subgroups") != std::end(Vec))
     return true;
-  return Device.get_info<info::device::version>() >= "2.1";
+
+  if (Device.get_backend() == sycl::backend::opencl) {
+    std::string ver = Device.get_info<info::device::version>();
+
+    // Extract the numerical version from the version string, OpenCL version
+    // string have the format "OpenCL <major>.<minor> <vendor specific data>".
+    return ver.substr(7, 3) >= "2.1";
+  }
+
+  return false;
```

Comment:
The change to the device version stuff looks good to me, but I'm a bit confused -- do we currently return false for all the non-OpenCL backends?  Or do the NVIDIA and AMD backends report support for cl_khr_subgroups in their extensions list?

Reply:
Yeah, I think the initial intent was for backends supporting it to return `cl_khr_subgroups` in their extension string. And I believe the extra version check for OpenCL is that this extension became core in OpenCL 2.1, so the devices stopped reporting the extension despite supporting it.

In theory we should probably report it from the Nvidia and AMD backends but I don't think we currently do. It's not ideal because since this is a runtime check it mostly looks like the tests are working even if they're disabled.

Reply:
Ok, that's pretty confusing.  Thanks for explaining it.

I think long-term we should look to remove this check entirely.  Sub-groups are a core feature of SYCL 2020 and should work everywhere.  But I'm happy for that to be done as part of a separate PR.